controllers/registeredservice_controller.go

```diff
@@ -45,6 +45,97 @@ func ServiceInCatalog(sc primazaiov1alpha1.ServiceCatalog, serviceName string) i
 	return -1
 }
 
+func (r *RegisteredServiceReconciler) RemoveServiceFromCatalog(ctx context.Context, catalogName string, namespace string, serviceName string) error {
```

Comment:
This function can be unexported

```suggestion
func (r *RegisteredServiceReconciler) removeServiceFromCatalog(ctx context.Context, catalogName string, namespace string, serviceName string) error {
```